clash-cores/src/Clash/Cores/Xilinx/Floating/Explicit.hs

```diff
@@ -311,6 +316,45 @@ fromS32 = withFrozenCallStack fromS32With
 -- | The default delay for conversion of @Signed 32@ to @Float@
 type FromS32DefDelay = 6
 
+-- | Compare two floating point numbers. Produces 'NaN' if any of the inputs
+-- is NaN.
+compareWith
+  :: forall d dom n
+   . ( KnownDomain dom
+     , KnownNat d
+     , HasCallStack
+     )
+  => Clock dom
+  -> Enable dom
+  -> DSignal dom n Float
+  -> DSignal dom n Float
+  -> DSignal dom (n + d) Ordering
+compareWith clk ena a b = delayI und ena clk (xilinxCompare <$> a <*> b)
+ where
+  und = withFrozenCallStack $ errorX "Initial values of compare undefined"
+{-# NOINLINE compareWith #-}
+{-# ANN compareWith (vhdlComparePrim 'compareWith 'compareTclTF "compare") #-}
+{-# ANN compareWith (veriComparePrim 'compareWith 'compareTclTF "compare") #-}
+
+-- | Compare two floating point numbers. Produces 'Nothing' if any of the inputs
+-- is NaN. Otherwise, it behaves like Haskell's 'P.compare'. Function has a
+-- default delay, unlike 'compareWith'.
```

Comment:
```suggestion
-- | Floating point comparison, with default delay
--
-- Produces 'NaN' if any of the inputs is NaN. Otherwise, it behaves like
-- Haskell's 'P.compare'.
```